src/site/content/en/blog/storage-foundation/index.md

```diff
@@ -0,0 +1,374 @@
+  creates a new file. Returns a promise that resolves with the the opened file.
+
+{% Aside 'warning' %} File names are restricted to lowercase alphanumeric characters and underscore
+(`a-z`, `0-9`, `_`). {% endAside %}
+
+{% Aside %} A file can only be opened once. This means concurrent access from different tabs is
+currently not possible. {% endAside %}
+
+- `storageFoundation.delete(name)`: Removes the file with the given name. Returns `undefined`.
+- `storageFoundation.rename(oldName, newName)`: Renames the file from the old name to the new name
+  atomically.  Returns `undefined`.
+- `storageFoundation.getAll()`: Returns a promise that resolves with an array of all existing file
+  names.
+- `storageFoundation.requestCapacity(requestedCapacity)`: Requests new capacity (in bytes) for usage
+  by the current execution context. Returns a promise that resolved with the remaining amount of
+  capacity available.
+
+{% Aside %} The Storage Foundation API achieves fast and predictable performance by implementing its
+own quota management system. Web applications must explicitly ask for capacity before storing any
+new data. This request will be granted according to the browser's quota guidelines. Anytime an
+application starts a new JavaScript execution context (e.g., new tab, new worker, reloading the
+page), it must make sure it owns sufficient capacity before writing new data. {% endAside %}
+
+- `storageFoundation.releaseCapacity(toBeReleasedCapacity)`: Releases unused capacity (in bytes)
+  from the current execution context. Returns a promised that resolves with the remaining amount of
+  capacity available.
+- `storageFoundation.getRemainingCapacity()`: Returns a promise that resolves with the capacity
+  available for the current execution context.
+
+### File handles
+
+Working with files happens via the following functions:
+
+{% Aside %} Storage Foundation API used to be called NativeIO. Some references to this name still
+remain and will be removed eventually. {% endAside %}
+
+- `NativeIOFile.close()`: Closes a file.  Returns `undefined`.
+- `NativeIOFile.flush()`: Synchronizes (that is, flushes) a file's in-core state with the storage
+  device.  Returns `undefined`.
+
+{% Aside %} It is a known issue that `flush()` might be slow and we are exploring whether offering a
+faster, less reliable variant would be useful. {% endAside %}
+
+- `NativeIOFile.getLength()`: Returns a promise that resolves with the length of the file in bytes.
+- `NativeIOFile.setLength(length)`: Sets the length of the file in bytes. If the new length is
+  smaller than the current length, bytes are removed starting from the end of the file. Otherwise
+  the file is extended with zero-valued bytes.  Returns `undefined`.
+- `NativeIOFile.read(buffer, offset)`: Reads the contents of the file at the given offset through a
+  buffer that is the result of transferring the given buffer, which is then left detached. Returns a
```

Comment:
I'm not sure I follow this. Are buffered file operations beyond the scope of this article? Maybe link to a discussion and call out that there's a deeper story here.

What does 'detached' mean? Does it mean that I can't write to the buffer after I've called `read()`?

Reply:
See the example:

```js
const readBuffer = new Uint8Array(3);
// Read at offset 1. `result.buffer` contains the transferred buffer,
// `result.readBytes` is 2, the number of bytes read. `readBuffer` is left
// detached.
result = await file.read(readBuffer, 1); 
// `Uint8Array(3) [65, 66, 0]`
```

It's a bit like an empty basket (`readBuffer`) that you take at the entrance of a store, put your groceries in, but then at the cashier your stuff gets moved to a bag (`result`) for you to take home.